ivy/functional/ivy/experimental/activations.py

```diff
@@ -14,21 +14,58 @@
     inputs_to_ivy_arrays,
     handle_device_shifting,
     handle_backend_invalid,
+    handle_complex_input,
 )
 
 
+def _logit_jax_like(
+    x: Union[float, int, ivy.Array],
+    /,
+    *,
+    fn_original: Optional[Callable] = None,
+    eps: Optional[float] = None,
+    out: Optional[ivy.Array] = None,
+):
+    real = ivy.real(x)
+    imag = ivy.imag(x)
+    if eps is None:
+        real = ivy.where(ivy.logical_or(real > 1, real < 0), ivy.nan, real)
+    else:
+        real = ivy.clip(real, eps, 1 - eps)
+    const = ivy.array(1j, dtype=x.dtype)
+    x = ivy.add(
+        real.astype(x.dtype),
+        ivy.multiply(
+            const,
+            imag.astype(x.dtype),
+        ),
+    )
+    output = ivy.log(
+        ivy.divide(
+            x,
+            ivy.subtract(
+                ivy.array(1).astype(x.dtype),
+                x,
+            ),
+        ),
+    )
+    return output
```

Comment:
This can most likely be simplified to leverage the existing backend implementations of logit. Why not use something like
```py
real = ivy.real(x)
imag = ivy.imag(x)
if eps is not None:
    real = ivy.clip(real, eps, 1-eps)
z = ivy.add(real, ivy.multiply(ivy.array(1j, dtype=x.dtype), imag))
return fn_original(z, eps=eps, out=out)
```

Reply:
you are always better than me at simplifying code.

Reply:
But I have one question why should we use fn_original my code handles if the real part is bigger than 1 or is smaller than zero  and also handle eps argument so we don't need the fn_original

Reply:
Ah, yeah, true. I was thinking to use `fn_original` to put the effort of actually doing the calculation on the backend, since they might have quicker implementations of `logit` than this function does. You make a good point about nans though, and looking at e.g. the paddle implementation it wouldn't really be able to handle that properly.

Okay, fair enough, you've convinced me. I do think it would be best to remove the `.astype`s though, since it should be fine without them if everything else is type safe (inside the subtract you can do `ivy.array(1., dtype=x.dtype)` instead of having a separate call to `.astype`)

Reply:
yeah sure.